client/state/plugins/installed/actions.js

```diff
@@ -51,6 +51,25 @@ import { getSite } from 'calypso/state/sites/selectors';
 
 import 'calypso/state/plugins/init';
 
+/**
+ * Determines the truthiness of a site specific property regardless of whether it is on the plugin object
+ * or on one of the plugin's site objects.
+ *
+ * @param {string} prop - The site property to check. One of 'active', 'autoupdate', 'update', or 'version'.
+ * @param {Object} plugin - The plugin object
+ * @param {number} siteId - The ID of the site if it exists
+ * @returns {boolean} True if the site specific property is on the plugin, false otherwise.
+ */
+const pluginHasTruthySiteProp = ( prop, plugin, siteId ) => {
+	if ( ! [ 'active', 'autoupdate', 'update', 'version' ].includes( prop ) ) {
+		throw new Error( `${ prop } is not a site property.` );
+	}
+
+	return plugin.hasOwnProperty( prop )
+		? plugin?.[ prop ]
+		: siteId && plugin.sites?.[ siteId ]?.[ prop ];
```

Comment:
Also, any chance `plugin.sites` will be undefined? That particular optional chaining might be unnecessary.

Reply:
Yes, it is undefined if the data object has come directly from the server. The `sites` prop is calculated on the client by the plugin selector.